File: Newbie_Contest/programmation_1.pl
# Newbie Contest - Challenge programmation 1

use WWW::Mechanize;

my $mech = WWW::Mechanize->new();

$mech->get( "https://www.newbiecontest.org" );

# Connexion
$mech->set_fields( "user" => [ 'kraken31', 1 ] );
$mech->set_fields( "passwrd" => [ 'dejean', 1 ] );
$mech->click();

# Récupération du nombre aléatoire
my $nombre = "";
$mech->get( "https://www.newbiecontest.org/epreuves/prog/prog1.php" );
my $page = $mech->content;

if ( $page =~ /.*: (\d*)/ ) {
	$nombre = $1;
}

# Récupération du mot de passe
my $password = "";
$mech->get( "https://www.newbiecontest.org/epreuves/prog/verifpr1.php?solution=$nombre" );
$page = $mech->content;
print $page;

File: Newbie_Contest/programmation_1.js
// Newbie Contest - Challenge programmation 1 avec CasperJS

var casper = require('casper').create();

var usernb = casper.cli.get(0); 
var passnb = casper.cli.get(1);

var number = "";
var url = "";
var password = "";

// Connexion au site
casper.start('https://www.newbiecontest.org', function() {
    casper.evaluate(function(user, pass) {
	    document.querySelector('#user').setAttribute('value', user);
	    document.querySelector('#passwrd').setAttribute('value', pass);
	    document.querySelector('form').submit();
	}, usernb, passnb);
});

// Récupération du nombre aléatoire
casper.thenOpen('https://www.newbiecontest.org/epreuves/prog/prog1.php', function() {
	if ( /.*: (\d*)/.test( this.getHTML() ) ) {
    	number = RegExp.$1;
    	
    	// Validation du nombre récupéré
    	url = 'https://www.newbiecontest.org/epreuves/prog/verifpr1.php?solution=' + number;
    	casper.open(url).then( function() {
    	
    		// Récupération du mot de passe
			if ( /.*: (\S*)/.test( this.fetchText('body') ) ) {
		    	password = RegExp.$1;
				this.echo("password="+password);
			}
    	});
    }
});

casper.run();
